app/create-app-template.js

```diff
@@ -16,6 +16,14 @@ const pathToId = file => {
 }
 
 module.exports = api => {
+  let entryExists = false
+  if (api.options.entry) {
+    try {
+      require.resolve(api.resolveBaseDir(api.options.entry))
+      entryExists = true
+    } catch (err) {}
+  }
```

Comment:
we can use `require.context` in the runtime code to check if there's entry file so that the app will be properly reloaded when you add a entry file.

Reply:
I added hot reload for the entry with `require.context`.

Since it requires string *literals* at compile time, the code is a bit verbose. I guess it could be moved into a reusable code generation function eventually (so the code becomes something like `const _entry = ${importOptional(api.baseDir, api.options.entry)}`, but I'm not entirely sure where you'd like it to reside.